Codechecker is too aggressive at removing some -include options

**Describe the bug**
Recently (since mid April), codechecker check started to fail with
`error: '-DDEBUG=1' file not found [clang-diagnostic-error]`

**CodeChecker version**
```
CodeChecker analyzer version:
---------------------------------------------------------------
Kind                 | Version                                 
---------------------------------------------------------------
Base package version | 6.10.0                                  
Package build date   | 2019-05-01T07:53                        
Git commit ID (hash) | 1fb4fa612017a51b783dceb1d42233b3cdc2690c
Git tag information  | 6.10                                    
---------------------------------------------------------------

CodeChecker web version:
------------------------------------------------------------------------
Kind                          | Version                                 
------------------------------------------------------------------------
Base package version          | 6.10.0                                  
Package build date            | 2019-05-01T07:53                        
Git commit ID (hash)          | 1fb4fa612017a51b783dceb1d42233b3cdc2690c
Git tag information           | 6.10                                    
Configuration schema          | v6.0                                    
Database schema               | v6.0                                    
Server supported API (Thrift) | 6.15                                    
Client API (Thrift)           | 6.15                                    
------------------------------------------------------------------------

Using clang-9
```

**To Reproduce**
Simple compilation db  - **compile_commands.json**

```

[
{
"directory": "/var/lib/jenkins/workspace/firefox-codechecker/obj-firefox/mfbt", 
"command": "/usr/bin/clang -std=gnu99 -o /dev/null -c -I/var/lib/jenkins/workspace/firefox-codechecker/obj-firefox/dist/system_wrappers -include /var/lib/jenkins/workspace/firefox-codechecker/config/gcc_hidden.h -DDEBUG=1 -Wno-gnu-zero-variadic-macro-arguments /var/lib/jenkins/workspace/firefox-codechecker/mfbt/lz4.c", 
"file": "/var/lib/jenkins/workspace/firefox-codechecker/mfbt/lz4.c"
}
]
```

```
$ CodeChecker check   --logfile compile_commands.json -o codechecker-results --enable google-readability-todo   
[INFO 2019-05-01 13:17] - Starting static analysis ...
[ERROR 2019-05-01 13:17] - Analyzing lz4.c with clangsa  failed!
[ERROR 2019-05-01 13:17] - 

[ERROR 2019-05-01 13:17] - 
<built-in>:1:10: fatal error: '-DDEBUG=1' file not found
#include "-DDEBUG=1"
         ^~~~~~~~~~~
1 error generated.

```

Note that running with --verbose debug, it seems that the cause might be the -include option:
`bj-firefox/dist/system_wrappers -include -DDEBUG=1 -Wno-`

**Expected behavior**
The analysis should be performed

--- a/analyzer/codechecker_analyzer/buildlog/log_parser.py
+++ b/analyzer/codechecker_analyzer/buildlog/log_parser.py
@@ -719,19 +719,6 @@ def __skip(flag_iterator, _):
     return False
 
 
-def __keep_except(substring):
-    def f(flag_iterator, details):
-        """
-        This function keeps the flag pointed by the given flag_iterator except
-        if it contains the given substring.
-        """
-        if substring not in flag_iterator.item:
-            details['analyzer_options'].append(flag_iterator.item)
-            return True
-        return False
-    return f
-
-
 def parse_options(compilation_db_entry, compiler_info_file=None):
     """
     This function parses a GCC compilation action and returns a BuildAction
@@ -771,24 +758,15 @@ def parse_options(compilation_db_entry, compiler_info_file=None):
     if '++' in details['compiler'] or 'cpp' in details['compiler']:
         details['lang'] = 'c++'
 
-    if 'clang' in details['compiler']:
-        flag_transformers = [
-            __skip,
-            __determine_action_type,
-            __skip_sources,
-            __get_language,
-            __get_output,
-            __keep_except(compilation_db_entry['file'])]
-    else:
-        flag_transformers = [
-            __skip,
-            __replace,
-            __collect_compile_opts,
-            __determine_action_type,
-            __skip_sources,
-            __get_arch,
-            __get_language,
-            __get_output]
+    flag_transformers = [
+        __skip,
+        __replace,
+        __collect_compile_opts,
+        __determine_action_type,
+        __skip_sources,
+        __get_arch,
+        __get_language,
+        __get_output]
 
     for it in OptionIterator(gcc_command[1:]):
         for flag_transformer in flag_transformers:

